<a href="https://colab.research.google.com/github/SmitM1/Coronary-Heart-Disease-Prediction-Using-Neural-Network/blob/main/Heart_Disease_Prediction_Using_Neural_Network.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression
from sklearn.neighbors import KNeighborsClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.linear_model import LogisticRegression
from xgboost import XGBClassifier
from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import mean_squared_error
from sklearn.metrics import mean_absolute_error
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
import tensorflow as tf
from tensorflow import keras
from keras.layers import Dense, Activation, Input
from keras.models import Sequential
from keras.src.losses import BinaryCrossentropy
from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score, roc_auc_score, confusion_matrix


In [ ]:
df = pd.read_csv('/content/CHDdata.csv')

In [ ]:
df

sbp  tobacco    ldl  adiposity  famhist  typea  obesity  alcohol  age  \
0    160    12.00   5.73      23.11  Present     49    25.30    97.20   52   
1    144     0.01   4.41      28.61   Absent     55    28.87     2.06   63   
2    118     0.08   3.48      32.28  Present     52    29.14     3.81   46   
3    170     7.50   6.41      38.03  Present     51    31.99    24.26   58   
4    134    13.60   3.50      27.78  Present     60    25.99    57.34   49   
..   ...      ...    ...        ...      ...    ...      ...      ...  ...   
457  214     0.40   5.98      31.72   Absent     64    28.45     0.00   58   
458  182     4.20   4.41      32.10   Absent     52    28.61    18.72   52   
459  108     3.00   1.59      15.23   Absent     40    20.09    26.64   55   
460  118     5.40  11.61      30.79   Absent     64    27.35    23.97   40   
461  132     0.00   4.82      33.41  Present     62    14.70     0.00   46   

     chd  
0      1  
1      1  
2      0  
3      1  
4      1  
..   ...  
457    0  
458    1  
459    0  
460    0  
461    1  

[462 rows x 10 columns]

In [ ]:
df.describe()

sbp     tobacco         ldl   adiposity       typea     obesity  \
count  462.000000  462.000000  462.000000  462.000000  462.000000  462.000000   
mean   138.326840    3.635649    4.740325   25.406732   53.103896   26.044113   
std     20.496317    4.593024    2.070909    7.780699    9.817534    4.213680   
min    101.000000    0.000000    0.980000    6.740000   13.000000   14.700000   
25%    124.000000    0.052500    3.282500   19.775000   47.000000   22.985000   
50%    134.000000    2.000000    4.340000   26.115000   53.000000   25.805000   
75%    148.000000    5.500000    5.790000   31.227500   60.000000   28.497500   
max    218.000000   31.200000   15.330000   42.490000   78.000000   46.580000   

          alcohol         age         chd  
count  462.000000  462.000000  462.000000  
mean    17.044394   42.816017    0.346320  
std     24.481059   14.608956    0.476313  
min      0.000000   15.000000    0.000000  
25%      0.510000   31.000000    0.000000  
50%      7.510000   45.000000    0.000000  
75%     23.892500   55.000000    1.000000  
max    147.190000   64.000000    1.000000

In [ ]:
y = df['chd']
X = df.drop(['chd'], axis=1)

In [ ]:
X

sbp  tobacco    ldl  adiposity  famhist  typea  obesity  alcohol  age
0    160    12.00   5.73      23.11  Present     49    25.30    97.20   52
1    144     0.01   4.41      28.61   Absent     55    28.87     2.06   63
2    118     0.08   3.48      32.28  Present     52    29.14     3.81   46
3    170     7.50   6.41      38.03  Present     51    31.99    24.26   58
4    134    13.60   3.50      27.78  Present     60    25.99    57.34   49
..   ...      ...    ...        ...      ...    ...      ...      ...  ...
457  214     0.40   5.98      31.72   Absent     64    28.45     0.00   58
458  182     4.20   4.41      32.10   Absent     52    28.61    18.72   52
459  108     3.00   1.59      15.23   Absent     40    20.09    26.64   55
460  118     5.40  11.61      30.79   Absent     64    27.35    23.97   40
461  132     0.00   4.82      33.41  Present     62    14.70     0.00   46

[462 rows x 9 columns]

In [ ]:
y

0      1
1      1
2      0
3      1
4      1
      ..
457    0
458    1
459    0
460    0
461    1
Name: chd, Length: 462, dtype: int64

In [ ]:
X['famhist'] = X['famhist'].replace({'Present':1, 'Absent':0})

In [ ]:
X

sbp  tobacco    ldl  adiposity  famhist  typea  obesity  alcohol  age
0    160    12.00   5.73      23.11        1     49    25.30    97.20   52
1    144     0.01   4.41      28.61        0     55    28.87     2.06   63
2    118     0.08   3.48      32.28        1     52    29.14     3.81   46
3    170     7.50   6.41      38.03        1     51    31.99    24.26   58
4    134    13.60   3.50      27.78        1     60    25.99    57.34   49
..   ...      ...    ...        ...      ...    ...      ...      ...  ...
457  214     0.40   5.98      31.72        0     64    28.45     0.00   58
458  182     4.20   4.41      32.10        0     52    28.61    18.72   52
459  108     3.00   1.59      15.23        0     40    20.09    26.64   55
460  118     5.40  11.61      30.79        0     64    27.35    23.97   40
461  132     0.00   4.82      33.41        1     62    14.70     0.00   46

[462 rows x 9 columns]

In [ ]:
y

0      1
1      1
2      0
3      1
4      1
      ..
457    0
458    1
459    0
460    0
461    1
Name: chd, Length: 462, dtype: int64

In [ ]:
scaler = StandardScaler()

In [ ]:
columns = X.columns

In [ ]:
columns

Index(['sbp', 'tobacco', 'ldl', 'adiposity', 'famhist', 'typea', 'obesity',
       'alcohol', 'age'],
      dtype='object')

In [ ]:
X = scaler.fit_transform(X)

In [ ]:
X

array([[ 1.05856354,  1.82307289,  0.47841217, ..., -0.17678588,
         3.27773797,  0.62933573],
       [ 0.27708929, -0.79023744, -0.15967999, ...,  0.67137291,
        -0.61274463,  1.3831147 ],
       [-0.99280637, -0.77498042, -0.60924492, ...,  0.73551937,
        -0.5411833 ,  0.21818356],
       ...,
       [-1.48122778, -0.13854454, -1.52287687, ..., -1.41457504,
         0.39238531,  0.83491181],
       [-0.99280637,  0.38455344,  3.32082269, ...,  0.31025208,
         0.28320317, -0.1929686 ],
       [-0.3090164 , -0.79241702,  0.0385153 , ..., -2.69512851,
        -0.69698253,  0.21818356]])

In [ ]:
X = pd.DataFrame(X, columns=columns)

In [ ]:
X

sbp   tobacco       ldl  adiposity   famhist     typea   obesity  \
0    1.058564  1.823073  0.478412  -0.295503  1.185854 -0.418470 -0.176786   
1    0.277089 -0.790237 -0.159680   0.412140 -0.843274  0.193344  0.671373   
2   -0.992806 -0.774980 -0.609245   0.884332  1.185854 -0.112563  0.735519   
3    1.546985  0.842264  0.807126   1.624141  1.185854 -0.214532  1.412621   
4   -0.211332  2.171805 -0.599577   0.305351  1.185854  0.703189 -0.012856   
..        ...       ...       ...        ...       ...       ...       ...   
457  3.696039 -0.705234  0.599263   0.812281 -0.843274  1.111065  0.571590   
458  2.133091  0.123004 -0.159680   0.861173 -0.843274 -0.112563  0.609602   
459 -1.481228 -0.138545 -1.522877  -1.309364 -0.843274 -1.336191 -1.414575   
460 -0.992806  0.384553  3.320823   0.692625 -0.843274  1.111065  0.310252   
461 -0.309016 -0.792417  0.038515   1.029720  1.185854  0.907127 -2.695129   

      alcohol       age  
0    3.277738  0.629336  
1   -0.612745  1.383115  
2   -0.541183  0.218184  
3    0.295062  1.040488  
4    1.647775  0.423760  
..        ...       ...  
457 -0.696983  1.040488  
458  0.068519  0.629336  
459  0.392385  0.834912  
460  0.283203 -0.192969  
461 -0.696983  0.218184  

[462 rows x 9 columns]

In [ ]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2)

In [ ]:
X_train

sbp   tobacco       ldl  adiposity   famhist     typea   obesity  \
192 -1.774281 -0.705234 -0.643083  -1.053325  1.185854  0.295313 -0.583047   
375 -0.602069 -0.792417  0.265715   0.287338 -0.843274 -2.865726  0.374399   
373 -0.504385  0.332244  0.077188   0.764676  1.185854  0.397282  0.089303   
272 -0.113648 -0.495995 -0.763934  -1.342816  1.185854  0.295313 -0.252811   
455 -0.504385 -0.304192 -0.923457   0.138089 -0.843274 -0.520439 -0.495142   
..        ...       ...       ...        ...       ...       ...       ...   
432 -0.113648 -0.792417 -0.357875  -0.816586 -0.843274 -1.336191 -0.975053   
203 -0.504385 -0.633308 -0.372377  -0.242752 -0.843274  0.091375 -0.530779   
51  -0.211332 -0.247523 -0.522232   0.706778 -0.843274 -0.112563  0.272239   
174 -0.504385 -0.792417 -1.078146   0.507351  1.185854 -0.010594 -0.948919   
143 -0.406701  0.188392  0.541255   1.546944 -0.843274  0.805158  1.227309   

      alcohol       age  
192 -0.612745 -0.261494  
375 -0.612745  0.149658  
373 -0.696983  1.451640  
272 -0.398470 -1.289374  
455  1.242125 -1.083798  
..        ...       ...  
432 -0.612745 -1.837577  
203  0.088147  1.451640  
51   0.270527  0.423760  
174 -0.640960  1.314589  
143  0.623835  1.040488  

[369 rows x 9 columns]

In [ ]:
X_test

sbp   tobacco       ldl  adiposity   famhist     typea   obesity  \
8   -1.188175 -0.792417 -0.440054  -0.772841  1.185854 -0.418470 -0.281321   
180  1.351616 -0.792417 -0.208020   1.140370 -0.843274 -0.826346  0.973099   
140  1.058564  2.258988  0.560591   1.507058 -0.843274  0.499251  1.859270   
46   1.156248  0.820468  1.841609  -0.097363  1.185854  1.111065 -0.079378   
245 -0.504385 -0.792417 -1.020138  -0.196433 -0.843274 -0.826346 -1.058206   
..        ...       ...       ...        ...       ...       ...       ...   
359  0.667826 -0.426248 -0.560905   0.002994 -0.843274 -0.316501  0.234227   
223 -0.113648  0.842264  1.280861   0.338803  1.185854 -0.316501 -0.245684   
424  0.179405 -0.312911 -0.701092  -0.348255 -0.843274 -0.928315 -0.566416   
70  -0.992806 -0.792417 -0.517398  -1.708217 -0.843274 -0.214532 -1.637900   
250  0.960879  0.552381  1.633746   0.687478 -0.843274 -0.724377  0.426666   

      alcohol       age  
8   -0.595161 -0.946748  
180 -0.154343  0.903437  
140 -0.553042  0.766386  
46  -0.457354  1.040488  
245 -0.429548  0.971963  
..        ...       ...  
359 -0.696983 -0.741171  
223 -0.696983  0.149658  
424 -0.465533 -0.055918  
70  -0.672447 -1.906103  
250  3.090452  0.355234  

[93 rows x 9 columns]

In [ ]:
y_train

192    1
375    0
373    0
272    0
455    1
      ..
432    0
203    0
51     0
174    0
143    0
Name: chd, Length: 369, dtype: int64

In [ ]:
y_test

8      0
180    0
140    1
46     1
245    0
      ..
359    0
223    1
424    1
70     0
250    0
Name: chd, Length: 93, dtype: int64

In [ ]:
model_nn = Sequential([
    Input(shape=(X_train.shape[1],)),
    Dense(64, activation='relu'),
    Dense(32, activation='relu'),
    Dense(1, activation='sigmoid')
])

In [ ]:
model_nn.compile(
    optimizer=keras.optimizers.Adam(learning_rate=0.01),
    loss=BinaryCrossentropy(),
    metrics=['accuracy']
)

In [ ]:
model_nn.fit(X_train, y_train, epochs=32, batch_size=10)

Epoch 1/32
37/37 [==============================] - 3s 5ms/step - loss: 0.5616 - accuracy: 0.7154
Epoch 2/32
37/37 [==============================] - 0s 2ms/step - loss: 0.5048 - accuracy: 0.7507
Epoch 3/32
37/37 [==============================] - 0s 2ms/step - loss: 0.4860 - accuracy: 0.7642
Epoch 4/32
37/37 [==============================] - 0s 2ms/step - loss: 0.4862 - accuracy: 0.7751
Epoch 5/32
37/37 [==============================] - 0s 2ms/step - loss: 0.4714 - accuracy: 0.7724
Epoch 6/32
37/37 [==============================] - 0s 2ms/step - loss: 0.4526 - accuracy: 0.7913
Epoch 7/32
37/37 [==============================] - 0s 2ms/step - loss: 0.4402 - accuracy: 0.7886
Epoch 8/32
37/37 [==============================] - 0s 2ms/step - loss: 0.4415 - accuracy: 0.8022
Epoch 9/32
37/37 [==============================] - 0s 2ms/step - loss: 0.4457 - accuracy: 0.7886
Epoch 10/32
37/37 [==============================] - 0s 2ms/step - loss: 0.4110 - accuracy: 0.7967
Epoch 11/32
37/37 [

In [ ]:
model = GaussianNB()

In [ ]:
model.fit(X_train, y_train)

GaussianNB()

In [ ]:
predictions_test = model.predict(X_test)

In [ ]:
predictions_test

array([0, 0, 1, 1, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0, 1, 0, 1, 0, 0,
       0, 1, 1, 0, 1, 0, 0, 0, 1, 1, 0, 0, 1, 0, 0, 1, 1, 1, 1, 0, 1, 0,
       0, 1, 1, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0,
       0, 1, 1, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1,
       0, 1, 0, 0, 1])

In [ ]:
predictions_test = [0 if x <= 0.5 else 1 for x in predictions_test]

In [ ]:
predictions_test

[0,
 0,
 1,
 1,
 0,
 1,
 1,
 0,
 0,
 1,
 0,
 0,
 0,
 0,
 1,
 1,
 0,
 1,
 0,
 1,
 0,
 0,
 0,
 1,
 1,
 0,
 1,
 0,
 0,
 0,
 1,
 1,
 0,
 0,
 1,
 0,
 0,
 1,
 1,
 1,
 1,
 0,
 1,
 0,
 0,
 1,
 1,
 1,
 1,
 0,
 0,
 0,
 0,
 0,
 1,
 0,
 0,
 0,
 0,
 0,
 1,
 0,
 1,
 1,
 0,
 0,
 0,
 1,
 1,
 1,
 0,
 0,
 1,
 1,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 1,
 0,
 1,
 1,
 0,
 1,
 0,
 0,
 1]

In [ ]:
predictions_train = model.predict(X_train)

In [ ]:
predictions_train

array([0, 0, 1, 0, 0, 1, 1, 1, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 1, 1, 0,
       0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 1, 1, 0, 1, 0, 1,
       0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0,
       0, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 1, 1,
       0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 1,
       0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1,
       0, 0, 1, 0, 0, 1, 1, 1, 1, 0, 1, 0, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1,
       1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0, 0,
       1, 0, 0, 0, 1, 1, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1,
       0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 1, 0, 1, 1, 1, 0, 0, 0,
       0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0,
       1, 1, 1, 0, 1, 0, 0, 1, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1, 1, 0, 1, 1,
       0, 1, 1, 1, 1, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 1, 0, 1, 1, 1, 0,
       0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 1, 1, 1, 1, 0,

In [ ]:
predictions_train = [0 if x <= 0.5 else 1 for x in predictions_train]

In [ ]:
predictions_train

[0,
 0,
 1,
 0,
 0,
 1,
 1,
 1,
 0,
 0,
 0,
 1,
 0,
 0,
 0,
 0,
 1,
 0,
 0,
 1,
 1,
 0,
 0,
 1,
 1,
 0,
 0,
 0,
 1,
 0,
 0,
 0,
 0,
 1,
 0,
 0,
 0,
 1,
 1,
 1,
 0,
 1,
 0,
 1,
 0,
 0,
 1,
 0,
 0,
 0,
 1,
 0,
 0,
 0,
 1,
 0,
 1,
 0,
 0,
 0,
 0,
 0,
 0,
 1,
 0,
 0,
 0,
 1,
 1,
 1,
 1,
 1,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 1,
 0,
 0,
 1,
 1,
 1,
 0,
 1,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 1,
 0,
 0,
 0,
 0,
 0,
 0,
 1,
 1,
 0,
 0,
 0,
 1,
 0,
 0,
 1,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 1,
 0,
 0,
 1,
 0,
 0,
 1,
 1,
 1,
 1,
 0,
 1,
 0,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 1,
 1,
 1,
 1,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 1,
 1,
 0,
 0,
 1,
 1,
 1,
 0,
 0,
 0,
 1,
 0,
 0,
 0,
 1,
 1,
 0,
 0,
 1,
 1,
 1,
 1,
 0,
 0,
 0,
 0,
 0,
 1,
 0,
 0,
 0,
 1,
 0,
 1,
 1,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 1,
 0,
 1,
 0,
 1,
 0,
 1,
 1,
 1,
 0,
 0,
 0,
 0,
 0,
 1,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 1,
 0,
 0,
 0,
 1,
 1,
 0,
 0,
 0,
 0,
 1,
 0,
 1,
 1,
 1,
 0,
 1,
 0,
 0,
 1,


In [ ]:
predictions_train = np.array(predictions_train)

In [ ]:
predictions_train

array([0, 0, 1, 0, 0, 1, 1, 1, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 1, 1, 0,
       0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 1, 1, 0, 1, 0, 1,
       0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0,
       0, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 1, 1,
       0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 1,
       0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1,
       0, 0, 1, 0, 0, 1, 1, 1, 1, 0, 1, 0, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1,
       1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0, 0,
       1, 0, 0, 0, 1, 1, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1,
       0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 1, 0, 1, 1, 1, 0, 0, 0,
       0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0,
       1, 1, 1, 0, 1, 0, 0, 1, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1, 1, 0, 1, 1,
       0, 1, 1, 1, 1, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 1, 0, 1, 1, 1, 0,
       0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 1, 1, 1, 1, 0,

In [ ]:
predictions_test = np.array(predictions_test)

In [ ]:
predictions_test

array([0, 0, 1, 1, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0, 1, 0, 1, 0, 0,
       0, 1, 1, 0, 1, 0, 0, 0, 1, 1, 0, 0, 1, 0, 0, 1, 1, 1, 1, 0, 1, 0,
       0, 1, 1, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0,
       0, 1, 1, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1,
       0, 1, 0, 0, 1])

In [ ]:
mse_train = mean_squared_error(y_train, predictions_train)
mse_train

0.2764227642276423

In [ ]:
mse_test = mean_squared_error(y_test, predictions_test)
mse_test

0.3010752688172043

In [ ]:
mae_train = mean_absolute_error(y_train, predictions_train)
mae_train

0.2764227642276423

In [ ]:
mae_test = mean_absolute_error(y_test, predictions_test)
mae_test

0.3010752688172043

In [ ]:
accuracy = accuracy_score(y_test, predictions_test)
print("Accuracy:", accuracy)

# Precision
precision = precision_score(y_test, predictions_test)
print("Precision:", precision)

# Recall
recall = recall_score(y_test, predictions_test)
print("Recall:", recall)

# F1-Score
f1 = f1_score(y_test, predictions_test)
print("F1-Score:", f1)

# ROC-AUC
roc_auc = roc_auc_score(y_test, predictions_test)
print("ROC-AUC:", roc_auc)

# Confusion Matrix
conf_matrix = confusion_matrix(y_test, predictions_test)
print("Confusion Matrix:")
print(conf_matrix)

Accuracy: 0.6989247311827957
Precision: 0.5789473684210527
Recall: 0.6470588235294118
F1-Score: 0.6111111111111113
ROC-AUC: 0.6879361914257228
Confusion Matrix:
[[43 16]
 [12 22]]


In [ ]:
accuracy = accuracy_score(y_train, predictions_train)
print("Accuracy:", accuracy)

# Precision
precision = precision_score(y_train, predictions_train)
print("Precision:", precision)

# Recall
recall = recall_score(y_train, predictions_train)
print("Recall:", recall)

# F1-Score
f1 = f1_score(y_train, predictions_train)
print("F1-Score:", f1)

# ROC-AUC
roc_auc = roc_auc_score(y_train, predictions_train)
print("ROC-AUC:", roc_auc)

# Confusion Matrix
conf_matrix = confusion_matrix(y_train, predictions_train)
print("Confusion Matrix:")
print(conf_matrix)

Accuracy: 0.7235772357723578
Precision: 0.5895522388059702
Recall: 0.626984126984127
F1-Score: 0.6076923076923078
ROC-AUC: 0.7003233392122281
Confusion Matrix:
[[188  55]
 [ 47  79]]


# Linear Regression

### Accuracy: 0.7506775067750677
### Precision: 0.6813186813186813
### Recall: 0.496
### F1-Score: 0.5740740740740741
### ROC-AUC: 0.6885737704918032
### Confusion Matrix:
[[215  29]
 [ 63  62]]

# K Nearest Neighbor Classification

### Accuracy: 0.7723577235772358
### Precision: 0.7368421052631579
### Recall: 0.5426356589147286
### F1-Score: 0.625
### ROC-AUC: 0.7192344961240311
### Confusion Matrix:
[[215  25]
 [ 59  70]]

# Random Forest Classification

### Accuracy: 0.7096774193548387
### Precision: 0.6086956521739131
### Recall: 0.4375
### F1-Score: 0.5090909090909091
### ROC-AUC: 0.6449795081967213
### Confusion Matrix:
[[52  9]
 [18 14]]

# Logistic Regression

### Accuracy: 0.7208672086720868
### Precision: 0.6296296296296297
### Recall: 0.5190839694656488
### F1-Score: 0.5690376569037656
### ROC-AUC: 0.6755083712874462
### Confusion Matrix:
[[198  40]
 [ 63  68]]

# XGBoost Classication

### Accuracy: 0.6774193548387096
### Precision: 0.56
### Recall: 0.42424242424242425
### F1-Score: 0.4827586206896552
### ROC-AUC: 0.6204545454545455
### Confusion Matrix:
[[49 11]
 [19 14]]

# Naive Bayes Classification

### Accuracy: 0.7235772357723578
### Precision: 0.5895522388059702
### Recall: 0.626984126984127
### F1-Score: 0.6076923076923078
### ROC-AUC: 0.7003233392122281
### Confusion Matrix:
[[188  55]
 [ 47  79]]